src/instructlab/model/download.py

```diff
@@ -48,38 +275,69 @@
 @click.pass_context
 @clickext.display_params
 def download(ctx, repository, release, filename, model_dir, hf_token):
-    """Download the model(s) to train"""
-    click.echo(f"Downloading model from {repository}@{release} to {model_dir}...")
-    if hf_token == "" and "instructlab" not in repository:
-        raise ValueError(
-            """HF_TOKEN var needs to be set in your environment to download HF Model.
-            Alternatively, the token can be passed with --hf-token flag.
-            The HF Token is used to authenticate your identity to the Hugging Face Hub."""
+    downloader = None
+
+    if is_oci_repo(repository):
+        downloader = OCIDownloader(
+            repository=repository, release=release, download_dest=model_dir, ctx=ctx
+        )
+    elif is_huggingface_repo(repository):
+        downloader = HFDownloader(
+            repository=repository,
+            release=release,
+            download_dest=model_dir,
+            filename=filename,
+            hf_token=hf_token,
+            ctx=ctx,
         )
+    else:
+        click.secho(
+            f"repository {repository} matches neither Hugging Face nor OCI registry format. Please supply a valid repository",
+            fg="red",
+        )
+        raise click.exceptions.Exit(1)
+
     try:
-        if ctx.obj is not None:
-            hf_logging.set_verbosity(ctx.obj.config.general.log_level.upper())
-        files = list_repo_files(repo_id=repository, token=hf_token)
-        if any(".safetensors" in string for string in files):
-            if not os.path.exists(os.path.join(model_dir, repository)):
-                os.makedirs(name=os.path.join(model_dir, repository), exist_ok=True)
-            snapshot_download(
-                token=hf_token,
-                repo_id=repository,
-                revision=release,
-                local_dir=os.path.join(model_dir, repository),
-            )
-        else:
-            hf_hub_download(
-                token=hf_token,
-                repo_id=repository,
-                revision=release,
-                filename=filename,
-                local_dir=model_dir,
-            )
+        downloader.download()
+    except FileNotFoundError as exc:
+        click.secho(
+            "skopeo is not installed, please install recommended version 1.15",
```

Comment:
This is a general function for any repo type (not just OCI). I think it's making an assumption that 1) OCI downloader will never raise the exception in any other scenario; 2) HF downloader will never raise it at all.

I think this `except` block belongs to where the `skopeo` presence is actually checked (inside `check_skopeo_version` or maybe inside the `OCIDownloader.download`).